pilot/pkg/networking/core/route/route.go

```diff
@@ -507,30 +507,32 @@ func TranslateRoute(
 	}
 
 	var hostnames []host.Name
-	if opts.InferencePoolExtensionRef != "" {
-		extSvc, extPort := strings.Split(opts.InferencePoolExtensionRef, ":")[0], strings.Split(opts.InferencePoolExtensionRef, ":")[1]
-		p, _ := strconv.ParseInt(extPort, 10, 32)
-		if out.TypedPerFilterConfig == nil {
-			out.TypedPerFilterConfig = make(map[string]*anypb.Any)
-		}
-		out.TypedPerFilterConfig[wellknown.HTTPExternalProcessing] = protoconv.MessageToAny(&extproc.ExtProcPerRoute{
-			Override: &extproc.ExtProcPerRoute_Overrides{
-				Overrides: &extproc.ExtProcOverrides{
-					GrpcService: &core.GrpcService{
-						TargetSpecifier: &core.GrpcService_EnvoyGrpc_{
-							EnvoyGrpc: &core.GrpcService_EnvoyGrpc{
-								ClusterName: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", host.Name(extSvc), int(p)),
+	if opts.InferencePoolExtensionRefs != nil {
+		if infPoolExtRef, ok := opts.InferencePoolExtensionRefs[in.Name]; ok {
```

Comment:
@howardjohn need a couple pair of eyes here - I should use in.Name and not routename - correct? httproute.Name in `CheckAndGetInferencePoolConfigs` should be before any mutations for the routeName